pyteal/ast/ecdsa.py

```diff
@@ -0,0 +1,169 @@
+from enum import Enum
+from typing import Tuple, TYPE_CHECKING
+
+from pyteal.ast import Expr, MultiValue
+from pyteal.errors import TealTypeError, verifyTealVersion
+from pyteal.ir import Op, TealBlock, TealOp
+from pyteal.types import TealType, require_type
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+EcdsaPubkey = [TealType.bytes, TealType.bytes]
+
+
+class EcdsaCurve(Enum):
+    """Enum representing an elliptic curve specification used in ECDSA."""
+
+    Secp256k1 = (0, "Secp256k1", 5)
+
+    def __init__(self, id: int, name: str, min_version: int) -> None:
+        self.id = id
+        self.arg_name = name
+        self.min_version = min_version
```

Comment:
nit: is there anywhere in the code checking the curve's min-version, or it is equal to `Ecdsa*` bundle of 3 opcodes' minimum version?

Reply:
Ah good catch.  At the moment, the latter is true but it may not be the case in the future if Secp256r1 is supported.

Reply:
(I think it is also the case when you use decompress and recover)